src/howler.core.js

```diff
@@ -438,6 +438,12 @@
       // Loop through the sources and pick the first one that is compatible.
       for (var i=0; i<self._src.length; i++) {
         var ext, str;
+        
+        // Make sure the source is actually a string
+        if (typeof str !== 'string') {
```

Comment:
`str` hasn't yet been defined here. The check should be:

```javascript
if (typeof self._src[i] !== 'string') {
```

Also, minor nitpick, but all of the comments have a period at the end (sorry, OCD kicking in).